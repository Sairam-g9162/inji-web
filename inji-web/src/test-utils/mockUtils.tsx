import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

const renderWithProvider = (ui: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {ui}
            </Router>
        </Provider>,
        options
    );
};

export * from '@testing-library/react';
export { renderWithProvider };
