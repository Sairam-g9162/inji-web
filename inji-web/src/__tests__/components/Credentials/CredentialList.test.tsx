import React from 'react';
import { CredentialList } from '../../../components/Credentials/CredentialList';
import { RequestStatus } from '../../../hooks/useFetch';
import { mockCredentials } from '../../../test-utils/mockObjects';
import { renderWithProvider } from '../../../test-utils/mockUtils'; // Import from mockutils

// Mock the i18n configuration
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: (key: string) => key,
    }),
    initReactI18next: {
        type: '3rdParty',
        init: jest.fn(),
    },
}));

// Mock the useSelector hook
jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
}));

describe("Test CredentialList Component", () => {
    beforeEach(() => {
        const useSelectorMock = require('react-redux').useSelector;
        useSelectorMock.mockImplementation((selector: any) => selector({
            credentials: {
                filtered_credentials: mockCredentials,
            },
            common: {
                language: 'en',
            },
        }));
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('matches the loading state snapshot', () => {
        const { asFragment } = renderWithProvider(<CredentialList state={RequestStatus.LOADING} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('matches the error state snapshot', () => {
        const { asFragment } = renderWithProvider(<CredentialList state={RequestStatus.ERROR} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('matches the empty credentials list snapshot', () => {
        const useSelectorMock = require('react-redux').useSelector;
        useSelectorMock.mockImplementation((selector: any) => selector({
            credentials: {
                filtered_credentials: []
            },
            common: {
                language: 'en',
            },
        }));

        const { asFragment } = renderWithProvider(<CredentialList state={RequestStatus.DONE} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('matches the credentials list snapshot', () => {
        const { asFragment } = renderWithProvider(<CredentialList state={RequestStatus.DONE} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
