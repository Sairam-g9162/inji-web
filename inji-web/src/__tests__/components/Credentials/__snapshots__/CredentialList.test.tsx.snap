// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CredentialList Component matches the credentials list snapshot 1`] = `
<DocumentFragment>
  <div
    class="font-bold text-iw-title mt-8 text-xl flex"
    data-testid="HeaderTile-Text"
  >
    containerHeading
  </div>
  <div
    class="text-iw-subTitle text-md mb-8 mt-2 flex"
    data-testid="HeaderTile-Text-subContent"
  >
    containerSubHeading
  </div>
  <div
    class="grid grid-cols-8 gap-6 md:grid-cols-6 sm:grid-cols-4"
  >
    <div
      class="bg-iw-tileBackground w-48 h-48 flex flex-col shadow hover:shadow-lg hover:scale-110 hover:shadow-iw-selectedShadow p-5 m-4 rounded-md cursor-pointer items-center"
      data-testid="ItemBox-Outer-Container-0"
      role="menuitem"
      tabindex="0"
    >
      <img
        alt="Issuer Logo"
        class="w-20 h-20 items-center justify-center me-4 flex"
        data-testid="ItemBox-Logo"
        src="https://url.com"
      />
      <div
        class="flex justify-center items-center pt-4"
      >
        <h3
          class="flex text-sm font-semibold text-iw-title text-center"
          data-testid="ItemBox-Text"
        >
          Name1
        </h3>
      </div>
    </div>
    <div
      class="bg-iw-tileBackground w-48 h-48 flex flex-col shadow hover:shadow-lg hover:scale-110 hover:shadow-iw-selectedShadow p-5 m-4 rounded-md cursor-pointer items-center"
      data-testid="ItemBox-Outer-Container-1"
      role="menuitem"
      tabindex="0"
    >
      <img
        alt="Issuer Logo"
        class="w-20 h-20 items-center justify-center me-4 flex"
        data-testid="ItemBox-Logo"
        src="https://url.com"
      />
      <div
        class="flex justify-center items-center pt-4"
      >
        <h3
          class="flex text-sm font-semibold text-iw-title text-center"
          data-testid="ItemBox-Text"
        >
          Name2
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test CredentialList Component matches the empty credentials list snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="font-bold text-iw-title mt-8 text-xl flex"
      data-testid="HeaderTile-Text"
    >
      containerHeading
    </div>
    <div
      class="text-iw-subTitle text-md mb-8 mt-2 flex"
      data-testid="HeaderTile-Text-subContent"
    >
      containerSubHeading
    </div>
    <div
      class="flex justify-center items-center w-full mx-auto bg-iw-background
 my-auto flex-col h-72 shadow-md shadow-iw-shadow border-t-iw-shadow px-10"
      data-testid="EmptyList-Outer-Container"
    >
      <p
        class="text-center text-light-title dark:text-dark-title"
        data-testid="EmptyList-Text"
      >
        emptyContainerContent
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test CredentialList Component matches the error state snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="font-bold text-iw-title mt-8 text-xl flex"
      data-testid="HeaderTile-Text"
    >
      containerHeading
    </div>
    <div
      class="text-iw-subTitle text-md mb-8 mt-2 flex"
      data-testid="HeaderTile-Text-subContent"
    >
      containerSubHeading
    </div>
    <div
      class="flex justify-center items-center w-full mx-auto bg-iw-background
 my-auto flex-col h-72 shadow-md shadow-iw-shadow border-t-iw-shadow px-10"
      data-testid="EmptyList-Outer-Container"
    >
      <p
        class="text-center text-light-title dark:text-dark-title"
        data-testid="EmptyList-Text"
      >
        emptyContainerContent
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test CredentialList Component matches the loading state snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center"
    data-testid="SpinningLoader-Container"
  >
    <div
      aria-busy="true"
      aria-label="oval-loading"
      class="sc-beySPh gZmPOF"
      data-testid="oval-loading"
      role="progressbar"
    >
      <svg
        data-testid="oval-svg"
        height="80"
        stroke="var(--iw-color-spinningLoaderPrimary)"
        viewBox="-20.5 -20.5 43 43"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            data-testid="oval-secondary-group"
            stroke-width="3"
            transform="translate(1 1)"
          >
            <circle
              cx="0"
              cy="0"
              r="20"
              stroke="var(--iw-color-spinningLoaderSecondary)"
              stroke-opacity=".5"
              stroke-width="3"
            />
            <path
              d="M20 0c0-9.94-8.06-20-20-20"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 0 0"
                repeatCount="indefinite"
                to="360 0 0"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
